toml2json/default.nix

```diff
@@ -18,7 +18,7 @@ in stdenv.mkDerivation {
 
   sourceRoot = ".";
 
-  nativeBuildInputs = [ rustc ];
+  nativeBuildInputs = [ rustc ] ++ lib.optionals stdenv.isDarwin [ darwin.libiconv ];
```

Comment:
I believe it's a link dependency, so it goes `buildInputs`.
Also you could use (`optional`, no `s`) `lib.optional stdenv.isDarwin darwin.libiconv`